hivemind/p2p/p2p_daemon_bindings/datastructures.py

```diff
@@ -0,0 +1,186 @@
+import hashlib
+from typing import Union, List, Sequence, Any
+
+import base58
+import multihash
+
+from multiaddr import Multiaddr, protocols
+from hivemind.proto import p2pd_pb2
+
+from hivemind.p2p.p2p_daemon_bindings.keys import PublicKey
+
+# NOTE: On inlining...
+# See: https://github.com/libp2p/specs/issues/138
+# NOTE: enabling to be interoperable w/ the Go implementation
+ENABLE_INLINING = True
+MAX_INLINE_KEY_LENGTH = 42
+
+IDENTITY_MULTIHASH_CODE = 0x00
+
+if ENABLE_INLINING:
+
+    class IdentityHash:
+        _digest: bytes
+
+        def __init__(self) -> None:
+            self._digest = bytearray()
+
+        def update(self, input: bytes) -> None:
+            self._digest += input
+
+        def digest(self) -> bytes:
+            return self._digest
+
+    multihash.FuncReg.register(
+        IDENTITY_MULTIHASH_CODE, "identity", hash_new=lambda: IdentityHash()
+    )
```

Comment:
```suggestion
    multihash.FuncReg.register(
        IDENTITY_MULTIHASH_CODE, "identity", hash_new=IdentityHash)
```

nit: `lambda` seems redundant.